### ISSUE TITLE
Detect when user supplied path instead of pattern

### ISSUE THREAD

**Issue by sharkdp** [sharkdp/fd#235]
From time to time it happens to me that I forget to supply a regex-pattern when using an explicit search path. In particular, this happens when I use `--extension EXT`/`-e EXT` because it feels like `EXT` is a pattern by itself:
``` bash
> fd -e jpg /share/pictures    # this will not do what you expect
```

The right way would be to use `fd -e jpg . /share/pictures` or `fd -e jpg "" /share/pictures`.

Instead, what the command above will try to do is to search for files in the current directory that match the regular expression `/share/pictures` (there will be none). I was wondering whether we could detect such cases and warn the user.

A possible implementation could look like this:

If all of the following conditions are fulfilled:
- only a single argument is given (which fd normally interprets as a search pattern)
- the pattern contains a path-separation character (`/` or `\`)
- (the *pattern* is an existing path in the filesystem)
- `--full-path`/`-p` has *not* been enabled (otherwise, we could potentially find entries that match the pattern)

... print a warning:
```
The search pattern "/share/pictures" looks like a path. If you want to specify a
root directory, use an empty search pattern: `fd [options] "" /share/pictures`.
```

**Comment by sharkdp** [comment]
In addition, we could also suggest `--full-path`/`-p` if somebody really wants to search for a part of a path.

**Comment by sharkdp** [comment]
Released as part of [fd v7.0.0](https://github.com/sharkdp/fd/releases/tag/v7.0.0)

**Comment by cup** [comment]
https://github.com/sharkdp/fd/issues/235
https://github.com/sharkdp/fd/issues/277
https://github.com/sharkdp/fd/issues/312
https://github.com/sharkdp/fd/issues/385
https://github.com/sharkdp/fd/pull/259
https://github.com/sharkdp/fd/pull/349
